chain/client/src/chunk_validation.rs

```diff
@@ -700,16 +700,26 @@ impl Client {
         }
 
         let chunk_header = chunk.cloned_header();
-        let chunk_validators = self
-            .epoch_manager
-            .get_chunk_validator_assignments(
-                epoch_id,
-                chunk_header.shard_id(),
-                chunk_header.height_created(),
-            )?
-            .ordered_chunk_validators();
+        let chunk_validators_assignment = self.epoch_manager.get_chunk_validator_assignments(
+            epoch_id,
+            chunk_header.shard_id(),
+            chunk_header.height_created(),
+        )?;
+        let mut chunk_validators = chunk_validators_assignment.ordered_chunk_validators();
         let witness =
             self.create_state_witness(prev_chunk_header, chunk, transactions_storage_proof)?;
+
+        if let Some(my_signer) = self.validator_signer.clone() {
+            // Since we've created the state witness, we can directly send the chunk endorsement to ourselves.
```

Comment:
but should we send the chunk endorsement to ourselves? Shouldn't we send it to the block producer?

Reply:
Absolute brain fart... Update the logic to send the block producers. I'm keeping the logic to send chunk endorsement to self tho, useful for tests